audit/audit.go

```diff
@@ -117,3 +120,56 @@ func (a *Audit) onLoggerError(err error) {
 		a.OnError(err)
 	}
 }
+
+// sortAuditFields sorts the context fields of an audit record such that some fields
+// are prepended in order, some are appended in order, and the rest are sorted alphabetically.
+// This is done to make reading the records easier since common fields will appear in the same order.
+func sortAuditFields(fields logr.Fields) []format.ContextField {
+	prependKeys := []string{KeyEvent, KeyStatus, KeyUserID, KeySessionID, KeyIPAddress}
+	appendKeys := []string{KeyClusterID, KeyClient}
+
+	// sort alphabetically any fields not in the prepend/append lists.
+	keys := make([]string, 0, len(fields))
+	for k := range fields {
```

Comment:
Given that fields are a `map[string]interface{}` we can leverage that and do something like:
```go
for k := range append(prependKeys, appendKeys...) {
    if _, ok := fields[k]; !ok {
        keys = append(keys, k)
    }
}
```

so we can remove the `findIn` method that and also beneficiate of a O(1) index, the caveat is to create a new slice.

Reply:
We need to check that each item's key from Fields map is not existing in either of two string arrays.  The algorithm you suggest checks the reverse - whether each string in the combined string arrays exists as a key in the fields map.

Reply:
Also, there won't be more than 10-20 items total.

Reply:
Nope is checking the `not ok`  instead of the ok so is checking if the field's key is not in those slices (preprendKeys and appendkeys combined). But it was a suggestion, given that there are 1-20 items total, to improve readability

Reply:
```go
for k := range append(prependKeys, appendKeys...) {
    if _, ok := fields[k]; !ok {
        keys = append(keys, k)
    }
}
```
> is checking if the field's key is not in those slices

I might be missing your intention, and apologies if I am, but that code is checking if strings in the slices are not keys in the fields map, which is not what we are trying to do.

Reply:
Sorry, you're right. I was thinking that the reverse way of checking for an existing key will be the opposite.

There is no direct way to filter keys in a map :sweat: